packages/entities/entities-plugins/src/components/PluginSelect.vue

```diff
@@ -0,0 +1,497 @@
+        plugin.disabledMessage = props.disabledPlugins[pluginId]
+      }
+
+      if (existingEntityPlugins.value.includes(pluginId)) {
+        plugin.exists = true
+
+        if (!plugin.disabledMessage) {
+          plugin.disabledMessage = t('plugins.select.already_exists')
+        }
+      }
+
+      const groupName = plugin.group || t('plugins.select.misc_plugins')
+      let plugins = list[groupName]
+
+      if (!plugins) {
+        plugins = []
+      }
+
+      plugins.push(plugin)
+      plugins.sort(sortAlpha('name'))
+
+      list[groupName] = plugins
+
+      return list
+    }, {})
+}
+
+const availablePluginsUrl = computed((): string => {
+  let url = `${props.config.apiBaseUrl}${endpoints.select[props.config.app].availablePlugins}`
+
+  if (props.config.app === 'konnect') {
+    url = url.replace(/{controlPlaneId}/gi, props.config?.controlPlaneId || '')
+  } else if (props.config.app === 'kongManager') {
+    url = url.replace(/\/{workspace}/gi, props.config?.workspace ? `/${props.config.workspace}` : '')
+  }
+
+  return url
+})
+
+const fetchEntityPluginsUrl = computed((): string => {
+  if (props.config?.entityType && props.config?.entityId) {
+    let url = `${props.config.apiBaseUrl}${endpoints.list[props.config.app].forEntity}`
+
+    if (props.config.app === 'konnect') {
+      url = url.replace(/{controlPlaneId}/gi, props.config?.controlPlaneId || '')
+    } else if (props.config.app === 'kongManager') {
+      url = url.replace(/\/{workspace}/gi, props.config?.workspace ? `/${props.config.workspace}` : '')
+    }
+
+    return url
+      .replace(/{entityType}/gi, props.config.entityType || '')
+      .replace(/{entityId}/gi, props.config.entityId || '')
+  }
+
+  return ''
+})
+
+// race condition between fetch of available plugins and setting
+// disabled/ignored plugins
+// if disabled/ignored plugins changes before availablePlugins fetch
+// wait for fetch to finish and don't attempt to build plugin list yet
+// if disabled/ignored plugins changes AFTER availablePlugins fetch
+// rebuild the list
+watch(() => props.disabledPlugins, (val, oldVal) => {
+  if (!objectsAreEqual(val, oldVal) && !isLoading.value) {
+    pluginsList.value = buildPluginList()
+  }
+})
+
+watch(() => props.ignoredPlugins, (val, oldVal) => {
+  if (!objectsAreEqual(val, oldVal) && !isLoading.value) {
+    pluginsList.value = buildPluginList()
+  }
+})
+
+const userCanCreateCustom = ref(false)
+const userCanEditCustom = ref(false)
+const userCanDeleteCustom = ref(false)
+onBeforeMount(async () => {
+  // Evaluate the user permissions
+  userCanCreateCustom.value = await props.canCreateCustom()
+  userCanEditCustom.value = await props.canEditCustom()
+  userCanDeleteCustom.value = await props.canDeleteCustom()
+})
+
+onMounted(async () => {
+  isLoading.value = true
+  hasError.value = false
+  fetchErrorMessage.value = ''
+
+  emit('loading', isLoading.value)
+
+  try {
+    const res = await axiosInstance.get(availablePluginsUrl.value)
```

Comment:
suggestion: since you're only accessing data, go ahead and destructure
```suggestion
    const { data } = await axiosInstance.get(availablePluginsUrl.value)
```